File: Code/SimpleBudget.API/Models/Taxes/TaxModel.cs
namespace SimpleBudget.API.Models
{
    public class TaxModel
    {
        public List<int> Years { get; set; } = default!;
        public List<TaxPersonModel> Persons { get; set; } = default!;
        public List<TaxIncomeModel> Incomes { get; set; } = default!;
        public List<TaxItemModel> TaxItems { get; set; } = default!;

        public int? SelectedPersonId { get; set; }
        public int SelectedYear { get; set; }
        public bool CanOpen { get; set; }

        public string? Closed { get; set; }

        public string ValueFormatCAD { get; set; } = default!;
    }
}

File: Code/client/src/api/models/TaxModel.ts
import { TaxIncomeModel } from "./TaxIncomeModel";
import { TaxItemModel } from "./TaxItemModel";
import { TaxPersonModel } from "./TaxPersonModel";

export interface TaxModel {
    years: number[];
    persons: TaxPersonModel[];
    incomes: TaxIncomeModel[];
    taxItems: TaxItemModel[];
    selectedPersonId: number | undefined | null;
    selectedYear: number;
    canOpen: boolean;
    closed: string | undefined | null;
    valueFormatCAD: string;
}
